src/overlays/actors/ovl_Boss_02/z_boss_02.c

```diff
@@ -97,186 +151,2144 @@ const ActorInit Boss_02_InitVars = {
+                if (this->unk_0158 > 20) {
+                    this->unk_015C = 0;
+                }
+                this->unk_0156 |= 4;
+                break;
+
+            case 10:
+                this->unk_015A = 21;
+                this->unk_0158 = this->unk_015A;
+                this->unk_015C++;
+
+            case 11:
+                this->unk_0158 -= 2;
+                if (this->unk_0158 <= 0) {
+                    this->unk_0158 = 0;
+                    this->unk_015C++;
+                }
+                this->unk_0156 |= 4;
+                break;
+
+            case 12:
+                this->unk_015A -= 2;
+                if (this->unk_015A <= 0) {
+                    this->unk_015A = 0;
+                    this->unk_015C = 0;
+                }
+                this->unk_0156 |= 4;
+                break;
+        }
+    }
+}
+
+void func_809DC78C(Actor* thisx, GlobalContext* globalCtx) {
+    Boss02* this = THIS;
+
+    this->unk_01AC = D_809DF5B0;
+    globalCtx->envCtx.unk_EA = 0xD;
+
+    if (D_809E0434 != NULL) {
+        this->unk_1D74 = KREG(23) + -15.0f;
+        D_801F4E30 = 0;
+        globalCtx->envCtx.unk_C3 = 1;
+        globalCtx->skyboxId = 1;
+    } else if (D_809E0422 == 0) {
+        this->unk_1D74 = 0.0f;
+        D_801F4E30 = this->unk_1D7C;
+        globalCtx->envCtx.unk_C3 = 0;
+        globalCtx->skyboxId = 2;
+    } else {
+        this->unk_1D74 = KREG(23) + -15.0f;
+        D_801F4E30 = ((KREG(24) * 0.1f) + 1.0f) * this->unk_1D7C;
+        globalCtx->envCtx.unk_C3 = 1;
+        globalCtx->skyboxId = 1;
+    }
+
+    Math_ApproachS(&this->unk_1D7C, this->unk_1D7E, 1, 3);
+
+    if (this->unk_0146[0] != 0) {
+        this->unk_0146[0]--;
+    }
+
+    if (D_809E0434 == NULL) {
+        if (D_809E0430 != 0) {
+            D_809E0430--;
+            if (D_809E0430 == 0) {
+                func_801A89A8(0x801B);
+            }
+        }
+    }
+
+    func_809DD0CC(globalCtx);
+    func_809DEAC4(this, globalCtx);
+    func_809DD934(this, globalCtx);
+}
+
+#ifdef NON_MATCHING
+// gfx/matrix macro at the start
+void Boss02_Draw(Actor* thisx, GlobalContext* globalCtx2) {
+    static Gfx* D_809DFA9C[] = {
+        D_0600ECF0, D_0600EF90, D_0600F310, D_0600F690, D_0600FA10, D_0600FD90, D_06010110, D_06010490,
+        D_06010810, D_06010B90, D_06010F10, D_06011290, D_06011610, D_06011990, D_06011D10, D_06012090,
+        D_06012410, D_06012790, D_06012B10, D_06012E90, D_06013210, D_06013590,
+    };
+    static Vec3f D_809DFAF4 = { -10000.0f, -100000.0f, -100000.0f };
+    Boss02* this = THIS;
+    GlobalContext* globalCtx = globalCtx2;
+    s32 i;
+    s32 idx;
+    RSPMatrix* matrix;
+    Gfx* gfx;
+    s32 phi_v0;
+    f32 phi_f12;
+    f32 spAC;
+    f32 spA8;
+    f32 spA4;
+    f32 spA0;
+    f32 sp9C;
+    f32 sp98;
+
+    matrix = (RSPMatrix*)globalCtx->state.gfxCtx->polyOpa.d - 23;
+    gfx = globalCtx->state.gfxCtx->polyOpa.d - 184;
+    globalCtx->state.gfxCtx->polyOpa.d = gfx;
```

Comment:
This looks like the `GRAPH_ALLOC` malloc
```c
GRAPH_ALLOC(globalCtx->state.gfxCtx, sizeof(RSPMatrix) * N);
```
Also, the variable named `gfx` may not be a dlist. It is only used to be set on the segment 0x0D, which I _think_ it is mainly used for matrices (I may be misremembering tho).
Maybe this will help you to match this function.